src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -171,7 +171,17 @@ private Uni<Supplier<CommandResult>> executeCollectionCreation(QueryExecutor que
                     return Uni.createFrom().item(false);
                   }
                 });
-    return indexResult.onItem().transform(SchemaChangeResult::new);
+    return indexResult
+        .onItem()
+        .transform(
+            res -> {
+              if (!res) {
+                // table creation failure or index creation failure
+                return ErrorCode.COLLECTION_CREATION_ERROR.toApiException();
```

Comment:
I think error message should contain name of Collection for which creation failed (should be available from context right?)

Reply:
yes, we can get from the context.
But not sure about this, since it is indeed the createCollectionOperation, and request path already has the collection name. 
Clients should be aware of the collection name they are creating?

Reply:
I'd recommend putting the collection name in the error message, it doesn't add too much bulk to the error message and it can help provide context in cases where the client isn't able to identify the offending collection from their logs.

Reply:
Right, context of call is not always super obvious from, say, logs. So a little bit of contextual information can be quite useful to know what individual response is about.

Reply:
Thanks guys! got it.